data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh

```diff
@@ -1,18 +1,58 @@
 #!/usr/bin/env bash
 
 wait_for_ha_api() {
-    if [ "$BOOTSTRAP_INPLACE" = true ]
-    then
-	return 0
-    fi
+    while :
+    do
+        get_topology
+        retcode=$?
+        if [[ $retcode -eq 2 ]]
+        then
+            echo "topology is SingleReplica, no need to wait for API availability"
+            return 0
+        fi
+        if [[ $retcode -eq 3 ]]
+        then
+            ## HA topology, we can start the wait loop for API availability
+            break
+        fi
+
+        ## error happened, so let's retry after 5s
+        sleep 5
+    done
 
     echo "Waiting for at least 2 available IP addresses for the default/kubernetes service"
     while ! is_api_available
     do
-	sleep 5
+        sleep 5
     done
 }
 
+
+get_topology() {
+    output=$(oc --kubeconfig="$KUBECONFIG" get infrastructures cluster -o jsonpath='{.status.controlPlaneTopology}' 2>&1 )
```

Comment:
@sadasu this is the proper way to identiy single node topology, and this PR will unblock the SNO team